task/show-sbom-rhdh/0.1/show-sbom-rhdh.yaml

```diff
@@ -0,0 +1,60 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: show-sbom-rhdh
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: "docker"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "containers, rhtap"
+    task.results.format: application/text
+    task.results.key: LINK_TO_SBOM
+    task.output.location: results
+spec:
+  description: >-
+    Shows the Software Bill of Materials (SBOM) generated for the built image in CyloneDX JSON format.
+    The 'task.*' annotations are processed by Red Hat Developer Hub (RHDH) so that the log content can be rendered in its UI.
+  params:
+    - name: IMAGE_URL
+      description: Fully qualified image name to show SBOM for.
+      type: string
+  results:
+  - description: Placeholder result meant to make RHDH identify this task as the producer of the SBOM logs.
+    name: LINK_TO_SBOM
+  steps:
+  - name: annotate-task
+    image: registry.redhat.io/openshift4/ose-cli:4.13@sha256:73df37794ffff7de1101016c23dc623e4990810390ebdabcbbfa065214352c7c
+    script: |
+      #!/usr/bin/env bash
+      echo "show-sbom-rhdh $(context.taskRun.name)"
+
+      # When this task is used in a pipelineRun triggered by Pipelines as Code, the annotations will be cleared,
+      # so we're re-adding them here
+      oc annotate taskrun $(context.taskRun.name) task.results.format=application/text
+      oc annotate taskrun $(context.taskRun.name) task.results.key=LINK_TO_SBOM
+      oc annotate taskrun $(context.taskRun.name) task.output.location=logs
+  - name: show-sbom
+    image: quay.io/redhat-appstudio/cosign:v2.1.1@sha256:c883d6f8d39148f2cea71bff4622d196d89df3e510f36c140c097b932f0dd5d5
```

Comment:
We should use the productized cosign (https://catalog.redhat.com/software/containers/rhtas-tech-preview/cosign-rhel9/652971cd584d77e9d922883b?architecture=amd64&image=65c5506e19b6942d637e5603)

Reply:
Yeah, but this image needs credentials to be pulled. I wasn' t able to test this in a full RHTAP install, so would this still work on it?

Reply:
It would, the [git-clone task works as well](https://github.com/redhat-appstudio/build-definitions/blob/3e548a38b3ad183262a25bc2a4eb6b5367b83fb5/task/git-clone/0.1/git-clone.yaml#L146). Tekton somehow magically has permissions to pull the images used to execute tasks from registry.redhat.io (maybe that's OpenShift magic)